kernel/src/fs/pipe.c

```
@@ -43,12 +58,16 @@ int pipe_create(FILE **read_fp, FILE **write_fp)
     read_node->flags = FS_PIPE;
     read_node->ptr = (fs_node_t *)pb;
     read_node->read = pipe_read_fn;
+    read_node->close = pipe_close_fn;
```

Cppcheck (2.21.0)

[warning] If memory allocation fails, then there is a possible null pointer dereference

(nullPointerOutOfMemory)